daemon/cluster/services.go

```diff
@@ -123,54 +131,61 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string) (*apity
 		if ctnr == nil {
 			return errors.New("service does not use container tasks")
 		}
-
 		if encodedAuth != "" {
 			ctnr.PullOptions = &swarmapi.ContainerSpec_PullOptions{RegistryAuth: encodedAuth}
 		}
 
-		// retrieve auth config from encoded auth
-		authConfig := &apitypes.AuthConfig{}
-		if encodedAuth != "" {
-			if err := json.NewDecoder(base64.NewDecoder(base64.URLEncoding, strings.NewReader(encodedAuth))).Decode(authConfig); err != nil {
-				logrus.Warnf("invalid authconfig: %v", err)
+		resp = &apitypes.ServiceCreateResponse{}
+
+		switch serviceSpec.Task.Runtime.(type) {
+		// handle other runtimes herer
+		case *swarmapi.TaskSpec_Container:
```

Comment:
The call above to `serviceSpec.Task.GetContainer()`, and related code, should move into this case body.

For readability, maybe move the body of this case to a separate function... but I'm not sure if there would be too many arguments for this to be practical.